### DISCUSSION TITLE
Bossbar not show if the players join after the command is executed

### DISCUSSION THREAD

**Discussion by bobhenl** [Zrips/CMI#8542] [Enhancement]
### Description of Issue

**EDIT, WHEN USING "ALL" INSTEAD PLAYERNAME, IT WORKS LIKE IT SHOULD, PLAYER SEE IT EVEN AFTER REJOIN. CANNOT THERE BE SOME SPECIAL FLAG TO MAKE IT WORK FOR SPECIFIC PLAYER AFTER REJOIN AS WELL, THE SAME WAY AS THE ALL INSTEAD OF PLAYER NAME DO NOW? BUT THE TEXT IS STILL A BIT RELEVANT IN CASE YOU REPLACE "ALL" WITH PLAYERNAME, IN ANOTHER SITUATIONS**

Hi, let' take this example https://www.zrips.net/bossbarmsg-handling/
`/cmi bossbarmsg all Server shutting down in [autoTimeLeft]! -cmd:"stop" -sec:-60 -c:red`

Is it possible to somehow execute this command even if the player joins later? Like imagine I set a cooldown for 10 minutes, then a bunch of players join between this time, and they won't see that there's a shutdown going on. I'd like to solve this somehow.

And there are many situations when you want to show the bossbar even when players join later, for example, for event, almost for everything where we can use bossbarmsg all it would be useful to display the text even when players join later. Maybe some new argument like -relog, which will allow displaying even after the player's relog, so until it ends?

Even if I put it into the eventcommands and execute it like with the time sec-60 all the time, I still need some placeholder that will show the right shutdown time, so I couldn't use  [autoTimeLeft] anymore, I'll need something like %countdown_600% but there doesn't seem to be any plugin like this. So, cannot this command be added to CMI - to set countdown, as alternative solution for this?

In this way I can just make eventcommand that'll execute this every time:
In eventcommands the command that will call the restartbossbarloop alias:
**ALTERNATIVE SOLUTION**
```
  restartbossbarloop:
    Cmds:
    - check:%countdown_600%<=0#!
    - asConsole! ph! /cmi bossbarmsg all Server shutting down in %countdown_600%! -cmd:"stop" -sec:-60 -c:red -cmd:"restartbossbarloop"
    Perm: true
    Tab: false
```
Like we can start countdown via command and then it can be checked via placeholder
Also note, this issue needs to be fixed too so it can work then I guess https://github.com/Zrips/CMI/issues/8528

So it will recursively call each time until it's solved. If there isn't a better solution, can you please add this countdown placeholder as it can solve this issue?

I thought about using the scheduler for this, as wel, but I'll still need some global cooldown placeholder which seems not to exist in any plugin

Or maybe /counter can be used for it somehow, but it seems that it miss the placeholder for actual counter value and not sure if it can work for this example

### Version Information

```shell
[17:34:24 INFO]: CMI: 9.6.10.7 Unknown  SqLite
[17:34:24 INFO]: CMILib: 1.4.5.1 -> 1.4.5.2
[17:34:24 INFO]: Server: Purpur(2062) 1.20.1-R0.1-SNAPSHOT-
[17:34:24 INFO]: CMI economy: Enabled Vault: 1.7.3-CMI CMI Chat: Disabled 
[17:34:24 INFO]: Modules -> 46 enabled 13 disabled: customMessages, mirror, tablist, skin, playerChatTag, deathMessages, disabledEnchants, ranks, firstJoinMessages, playerCombat, votifier, sitAnimation, namePlates
[17:34:24 INFO]: -----------------------------------------------
```

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by Zrips** [comment]
At this moment due to how bossbars being handled it will only show up for the players who are online at the moment command was performed. To have it show up for any new joining players would involve quite few changes and quite few improvements on entire system. This isn't really a bug, it's just limitation of the system. 
Suggestion for time being would be to lock server with something like "/cmi maintenance Server restarting, come back in few sec" which would prevent new joins while you have bossbar counting down. Not ideal especially if you have longer timers.
Will do some changes for CMILib to prepare for possible future improvements for bossbar being shown for everyone who joins after the fact, but for now its just limitation.

**Reply by bobhenl** [reply]
The main issue is that when you display some message to the player, and you want it to be like a counter or timer, then after relog, it's not displayed again. Would be nice if it can be looked into in the future. For example, I use it mainly for messages like the Damage protection etc. that is displayed for the player with counter from 1h down, but when they reconnect they don't know they have it still active.

As suggested, some placeholder that we can use can help too. Like some countdown placeholder.

@Zrips
